Cart item associations

connieelee opened the pull request:
closes #16

connieelee commented:
Created table that keeps track of items in a user's cart. Each user has a unique id and one cart, so the cart_item model links Squares, Users, and adds some additional info

zekenie commented on server/db/index.js:
```diff
@@ -2,8 +2,9 @@
 
-// if we had more models, we could associate them in this file
-// e.g. User.hasMany(Reports)
+CartItem.hasMany(Square, {as: 'product'});
```
this will mean your sequelize methods will be `getProducts` and `setProducts` not `setSquares` and `getSquares`. Not sure if this is what you want but it might be

zekenie commented on server/db/models/cart_item.js:
```diff
@@ -0,0 +1,20 @@
+}, {
+    hooks: {
+        beforeCreate: function (cart_item) {
+            cart_item.date_added = Sequelize.NOW;
```
this might be covered by the `createdAt` field already on every sequelize model

connieelee commented:
Updated cart_item model:
- removed date_added (can be represented by sequelize's "createdAt" and "updatedAt" attributes)
- association between cart items and squares renamed from "product" to "square"
